commit 7e8d12bc733fc9deeb5d78140cda169030fa2ac5
Author: Michał Winiarski <michal.winiarski@intel.com>
Date:   Fri Jan 20 20:23:46 2017 +0100

    drm/i915/huc: Avoid attempting to authenticate non-existent fw
    
    HuC authentication is called even if HuC firmware is not present in the
    system, leading to NULL ptr dereference on not allocated gem_object.
    Let's avoid trying to authenticate HuC if its firmware is not loaded
    successfully.
    
    Fixes: dac84a388528 ("drm/i915/huc: Support HuC authentication")
    
    v2: Check inside the auth function, split the assert (Michał)
    v3: Oops, hit send before compiling, s/huc_fw/huc->fw
    
    Cc: Anusha Srivatsa <anusha.srivatsa@intel.com>
    Cc: Arkadiusz Hiler <arkadiusz.hiler@intel.com>
    Cc: Chris Wilson <chris@chris-wilson.co.uk>
    Reviewed-by: Michal Wajdeczko <michal.wajdeczko@intel.com>
    Signed-off-by: Michał Winiarski <michal.winiarski@intel.com>
    Link: http://patchwork.freedesktop.org/patch/msgid/20170120192348.2049-1-michal.winiarski@intel.com
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/drivers/gpu/drm/i915/intel_huc.c b/drivers/gpu/drm/i915/intel_huc.c
index 897ef31d865e..5edd4f7c3c60 100644
--- a/drivers/gpu/drm/i915/intel_huc.c
+++ b/drivers/gpu/drm/i915/intel_huc.c
@@ -299,6 +299,9 @@ void intel_guc_auth_huc(struct drm_i915_private *dev_priv)
 	int ret;
 	u32 data[2];
 
+	if (huc->fw.load_status != INTEL_UC_FIRMWARE_SUCCESS)
+		return;
+
 	vma = i915_gem_object_ggtt_pin(huc->fw.obj, NULL, 0, 0,
 				PIN_OFFSET_BIAS | GUC_WOPCM_TOP);
 	if (IS_ERR(vma)) {